src/index.js

```diff
@@ -2,60 +2,80 @@
 /* eslint-disable prefer-destructuring */
 import './style.css';
 import Status from './task-status';
+import Crud from './tasks-handler';
 
-const dataStorage = JSON.parse(localStorage.getItem('todoTasks'));
-
-let task1 = {
-  description: 'Go to the supermarket',
-  completed: false,
-  index: 0,
-};
-
-let task2 = {
-  description: 'Dog Walking',
-  completed: false,
-  index: 1,
-};
-
-let task3 = {
-  description: 'Write some js code',
-  completed: false,
-  index: 2,
-};
-
-if (dataStorage !== null) {
-  task1 = dataStorage[0];
-  task2 = dataStorage[1];
-  task3 = dataStorage[2];
-}
-
-const toDoTasks = [task1, task2, task3];
-const taskList = document.querySelector('#todo-list');
+let toDoTasks = [];
 
 function setLocalStorage() {
   localStorage.setItem('todoTasks', JSON.stringify(toDoTasks));
 }
 
+const dataStorage = JSON.parse(localStorage.getItem('todoTasks'));
+
+const taskList = document.querySelector('#todo-list');
+const clearButton = document.querySelector('.link-button');
+
 function taskListDisplayed() {
+  if (dataStorage !== null) {
+    toDoTasks = dataStorage;
+  }
+  if (taskList.hasChildNodes()) {
+    const nodesArray = document.querySelectorAll('.task');
+    nodesArray.forEach((node) => {
+      taskList.removeChild(node);
+    });
+  }
   toDoTasks.forEach((task, index) => {
     const li = document.createElement('li');
     const labelDes = document.createElement('label');
     const checkBox = document.createElement('input');
     const icon = document.createElement('i');
+    const trashIcon = document.createElement('i');
 
     checkBox.type = 'checkbox';
     checkBox.classList.add('checkBox');
     checkBox.id = 'unmarked';
     li.classList.add('task');
+    li.id = index;
     icon.classList.add('fas');
     icon.classList.add('fa-ellipsis-v');
+    icon.id = 'move';
+    labelDes.contentEditable = false;
+    trashIcon.classList.add('fas');
+    trashIcon.classList.add('fa-trash-alt');
+    trashIcon.classList.add('hidden');
 
     labelDes.innerHTML = task.description;
 
     li.appendChild(checkBox);
     li.appendChild(labelDes);
     li.appendChild(icon);
-    taskList.appendChild(li);
+    li.appendChild(trashIcon);
+    taskList.insertBefore(li, clearButton);
+
+    li.addEventListener('click', () => {
+      labelDes.contentEditable = true;
+      if (trashIcon.classList.contains('hidden')) {
+        li.classList.add('task-pressed');
+        trashIcon.classList.remove('hidden');
+        icon.classList.add('hidden');
+      } else {
+        li.classList.remove('task-pressed');
+        icon.classList.remove('hidden');
+        trashIcon.classList.add('hidden');
+      }
+    });
+
+    labelDes.addEventListener('input', () => {
+      Crud.updateDescription(toDoTasks, li.id, labelDes.innerHTML);
```

Comment:
- [ ] I don't think an `innerHTML` can be passed as an input argument here. When the description is edited it assigns not just the string the user inputs but also the HTML elements too. You can see this effect on the attached screenshot on the localStorage. :pray: 

![Screenshot from 2021-09-24 01-20-46](https://user-images.githubusercontent.com/64894927/134571254-22f61312-a0b5-42e4-811f-b1f22fa4dab4.png)